### DISCUSSION TITLE
Mask input limit allows more characters than defined

### DISCUSSION THREAD

**Discussion by LTKort** [alpinejs/alpine#3868] [1. Help]
I want to have an input field with a limit of 4 characters, e.g. '9999'. If more than maximum characters are entered, the input should not register the extra input.

When using; `x-mask="9999"`, I can still type more than 4 characters, only the 5th is not displayed until the validation.

For example i create an input; 
`<input required="required" id="width" min="200" max="1800" type="number" x-mask="9999" wire:model.blur="width"  autocomplete="">`

Previously we used the plugin from  [ryangjchandler/alpine-mask](https://github.com/ryangjchandler/alpine-mask), which uses cleave.js.  If we used `x-mask.blocks="[4]"` it did work as intended.

Is this a bug with Alpine & Livewire? Or does it work as intended, and i need to change things, if so how?

Thanks!

**Comment by ekwoka** [comment]
Considering other similar bugs I've seen, I believe this to be an issue in Livewire that is accepting the change before the mask actually does anything.

If you use just x-model, you should only see the 4 characters used there.

**Comment by emaia** [comment]
I have the same problem. My input is:

```
<input
    wire:model="form.phone"
    x-mask="(99) 99999-9999"
    type="tel"
    placeholder="(--)"
    required />
```
Example: If I type the number nine 12 times , Livewire will try to validate "(99) 99999-99999" instead of "(99) 99999-9999". 
If I press backspace and press 5 one time, it works as expected: "(99) 99999-9999".

Minimal reproduction: https://wirebox.app/b/xp3kq

**Comment by emaia** [comment]
@LTKort adding debounce fix it for me: `wire:model.debounce.50ms`
